google-cloud-functions/google_api_auth.py

```diff
@@ -0,0 +1,152 @@
+"""
+Copyright start
+MIT License
+Copyright (c) 2024 Fortinet Inc
+Copyright end
+"""
+
+import requests
+from time import time, ctime
+from datetime import datetime
+from connectors.core.connector import get_logger, ConnectorError
+from connectors.core.utils import update_connnector_config
+
+logger = get_logger('google-cloud-functions')
+
+CONFIG_SUPPORTS_TOKEN = True
+
+REFRESH_TOKEN_FLAG = False
+
+# redirect url
+DEFAULT_REDIRECT_URL = 'https://localhost/myapp'
+
+# grant types
+REFRESH_TOKEN = 'refresh_token'
+
+
+class GoogleAuth:
+
+    def __init__(self, config):
+        self.client_id = config.get("client_id")
+        self.client_secret = config.get("client_secret")
+        self.verify_ssl = config.get('verify_ssl')
+        self.host = config.get("server")
+        if self.host[:7] == "http://":
+            self.host = "https://{0}".format(self.host)
+        elif self.host[:8] == "https://":
+            self.host = "{0}".format(self.host)
+        else:
+            self.host = "https://{0}".format(self.host)
+        self.refresh_token = ""
+        self.code = config.get("code")
+        self.token_url = 'https://accounts.google.com/o/oauth2/token'
+        if not config.get("redirect_url"):
+            self.redirect_url = DEFAULT_REDIRECT_URL
+        else:
+            self.redirect_url = config.get("redirect_url")
+
+    def convert_ts_epoch(self, ts):
+        datetime_object = datetime.strptime(ctime(ts), "%a %b %d %H:%M:%S %Y")
+        return datetime_object.timestamp()
+
+    def generate_token(self, REFRESH_TOKEN_FLAG):
```

Comment:
this function parameter name can be changed as there is constant declared in this file with same name